python/cudf/cudf/core/indexing.py

```diff
@@ -164,17 +164,23 @@ def __setitem__(self, key, value):
         self._sr.iloc[key] = value
 
     def _loc_to_iloc(self, arg):
-        from cudf.core.column import column
-        from cudf.core.series import Series
-
         if is_scalar(arg):
-            try:
-                found_index = self._sr.index._values.find_first_value(
-                    arg, closest=False
-                )
+            if isinstance(arg, int):
```

Comment:
I found a function that might be useful to check for valid integer objects: https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.api.types.is_integer.html#

In pandas `int`, and `np.integer` are ints, and bools are not ints. In python, `Bool` is `Int`.
```
>>> pd.api.types.is_integer(100)
True
>>> pd.api.types.is_integer(-100)
True
>>> pd.api.types.is_integer(np.uint16(10))
True
>>> pd.api.types.is_integer(np.int8(-42))
True
>>> pd.api.types.is_integer(True)
False
```

Reply:
Thank you! :))